code/edgeconsent/src/androidTest/java/com/adobe/marketing/mobile/edge/consent/ConsentBootUpTests.java

```diff
@@ -58,30 +69,42 @@ public void test_BootUp_loadsFromPersistence() throws Exception {
 		// verify consent response event dispatched
 		List<Event> consentResponseEvents = getDispatchedEventsWith(EventType.CONSENT, EventSource.RESPONSE_CONTENT);
 		assertEquals(1, consentResponseEvents.size());
-		Map<String, String> consentResponseData = flattenMap(consentResponseEvents.get(0).getEventData());
-		assertEquals(4, consentResponseData.size());
-		assertEquals("p", consentResponseData.get("consents.collect.val"));
-		assertEquals("n", consentResponseData.get("consents.adID.val"));
-		assertEquals("vi", consentResponseData.get("consents.personalize.content.val"));
-		assertEquals(SAMPLE_METADATA_TIMESTAMP, consentResponseData.get("consents.metadata.time"));
+		Map<String, Object> consentResponseData = consentResponseEvents.get(0).getEventData();
+
+		String expected =
+			"{" +
+			"  \"consents\": {" +
+			"    \"collect\": {" +
+			"      \"val\": \"p\"" +
+			"    }," +
+			"    \"adID\": {" +
+			"      \"val\": \"n\"" +
+			"    }," +
+			"    \"personalize\": {" +
+			"      \"content\": {" +
+			"        \"val\": \"vi\"" +
+			"      }" +
+			"    }," +
+			"    \"metadata\": {" +
+			"      \"time\": \"" +
+			SAMPLE_METADATA_TIMESTAMP +
+			"\"" +
+			"    }" +
+			"  }" +
+			"}";
+
+		JSONAsserts.assertEquals(expected, consentResponseData);
 
 		//  verify getConsent API
 		Map<String, Object> getConsentResponse = getConsentsSync();
-		Map<String, String> responseMap = flattenMap(
-			(Map) getConsentResponse.get(ConsentTestConstants.GetConsentHelper.VALUE)
-		);
-		assertEquals("p", responseMap.get("consents.collect.val"));
-		assertEquals("n", responseMap.get("consents.adID.val"));
-		assertEquals("vi", responseMap.get("consents.personalize.content.val"));
-		assertEquals(SAMPLE_METADATA_TIMESTAMP, responseMap.get("consents.metadata.time"));
+		Map<String, Object> responseMap = (Map) getConsentResponse.get(ConsentTestConstants.GetConsentHelper.VALUE);
+
+		JSONAsserts.assertEquals(expected, responseMap);
 
 		// verify xdm shared state
-		Map<String, String> xdmSharedState = flattenMap(getXDMSharedStateFor(ConsentConstants.EXTENSION_NAME, 2000));
-		assertEquals(4, xdmSharedState.size());
-		assertEquals("p", xdmSharedState.get("consents.collect.val"));
-		assertEquals("n", xdmSharedState.get("consents.adID.val"));
-		assertEquals("vi", xdmSharedState.get("consents.personalize.content.val"));
-		assertEquals(SAMPLE_METADATA_TIMESTAMP, xdmSharedState.get("consents.metadata.time"));
+		Map<String, Object> xdmSharedState = getXDMSharedStateFor(ConsentConstants.EXTENSION_NAME, 2000);
+
+		JSONAsserts.assertEquals(expected, xdmSharedState);
```

Comment:
@timkimadobe @cacheung  Do we need to assert on size since it might return true even if the sharedstate has more than 4 elements?